### PR TITLE
Map deployment strategy to eks

### LINKED ISSUE #359: Allow configuration of kubernetes deploy strategy
Carry forward (and possibly map?) the deploy strategy in mu.yml to the kubernetes deployment resource

https://container-solutions.com/kubernetes-deployment-strategies/

### PR DESCRIPTION
**Pull request by BennyHarv3**
Closes #359.

Functional testing of each strategy complete. There is an issue where you can't change your deployment strategy from blue_green or rolling to replace after the EKS cluster has been created (and likely vice versa, didn't check), but I don't expect it to be much of a problem. Changing from blue_green to rolling and back works fine.

### CODE REVIEW

**Comment by BennyHarv3** [comment]
As a heads up, whenever When #345 occurs, it causes the ELB generated by EKS to not be cleaned up properly. The workflow where I encountered it:

1.  Run `make e2e_eks` to start the test run
1.  Issue #345 occurs part way through the Deploy CodeBuild project.
1.  Manually enable the KMS key.
1.  Rerun the final section of the CodePipeline to get everything to pass.
1.  Run `mu purge`.

`mu purge` fails on the VPC stack because the VPC has dependencies. The only way I was able to get `mu purge` to continue on is manually deleting the ELB, manually deleting the VPC, and then manually deleting the CloudFormation stack.

File: workflows/service_deploy.go [hunk 1]
```diff
@@ -416,12 +447,12 @@ func (workflow *serviceWorkflow) serviceEksDBSecret(namespace string, service *c
 	}
 }
 
+// serviceEksDeployer accepts a service and its information and upserts a kubernetes Pod file to
+// a k8s cluster
 func (workflow *serviceWorkflow) serviceEksDeployer(namespace string, service *common.Service, stackParams map[string]string, environmentName string) Executor {
 	return func() error {
 		log.Noticef("Deploying service '%s' to '%s' from '%s'", workflow.serviceName, environmentName, workflow.serviceImage)
 
-		resolveServiceEnvironment(service, environmentName)
```
Comment (cplee) [inline workflows/service_deploy.go:455]: why is this removed?

Comment (BennyHarv3) [reply workflows/service_deploy.go:455]: It's called twice in that method. I didn't think the order mattered, so I removed the first one and left the second. If both calls are needed, or if the first one is the correct one, I'll put it back.

Comment (cplee) [reply workflows/service_deploy.go:455]: ha!  good catch.
